bin/library-detection.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Generates custom-metrics/third-parties.js using the tests from the 
+ * Library Detector project: https://github.com/johnmichel/Library-Detector-for-Chrome
+ */
+const fetch = require('node-fetch');
+const fsp = require('fs-promise');
+
+const LIBRARY_DETECTOR_URL = 'https://raw.githubusercontent.com/johnmichel/Library-Detector-for-Chrome/master/';
+const LIBRARY_DETECTOR_MAIN_URL = `${LIBRARY_DETECTOR_URL}/library/libraries.js`;
+const LIBRARY_DETECTOR_MANIFEST_URL = `${LIBRARY_DETECTOR_URL}/manifest.json`;
+const LIBRARY_DETECTOR_VAR_PATTERN = /([a-f0-9]*_LibraryDetectorTests)/;
+const TMP_FILE = 'tmp/libraryDetector.js';
+const VERSION_METRIC_FILE = 'custom_metrics/lib-detector-version.js';
+const OUTPUT_FILE = 'custom_metrics/third-parties.js';
+const OUTPUT_HEAD = `/**
+ * IMPORTANT: Do not modify this file directly! It is generated by
+ *   bin/library-detection.js
+ *
+ * Detects the presence of third party libraries.
+ *
+ * Look for common library aliases on the global scope.
+ * If available, detect libraries' versions.
+ * Built on https://github.com/johnmichel/Library-Detector-for-Chrome.
+ *
+ * Outputs JSON-serialized list of library-version pairs.
+ *     e.g.: \`["a@1.0","b","c@2.2"]\`
+ */
+var thirdParties = [];
+
+function addThirdParty(name, version) {
+	if (!name) {
+		return;
+	}
+
+	name = name.replace('@', '\\@');
+	version = version && version.replace('@', '\\@');
+	name += '@' + (version || 'null');
```

Comment:
I thought the split might be able to take a pattern like `[^\]@`. Also considered replacing `@` with `<at>` or similar.

I agree that it could be made more BQ friendly. For example, an array of objects similar to `{name: 'jQuery', version: '1.2.3'}`.